pkg/api/dashboard.go

```diff
@@ -124,16 +130,23 @@ func getUserLogin(userId int64) string {
 	}
 }
 
-func getDashboardHelper(orgId int64, slug string, id int64) (*m.Dashboard, Response) {
-	query := m.GetDashboardQuery{Slug: slug, Id: id, OrgId: orgId}
+func getDashboardHelper(orgId int64, slug string, id int64, uid string) (*m.Dashboard, Response) {
+	var query m.GetDashboardQuery
+
+	if len(uid) > 0 {
+		query = m.GetDashboardQuery{Uid: uid, Id: id, OrgId: orgId}
+	} else {
+		query = m.GetDashboardQuery{Slug: slug, Id: id, OrgId: orgId}
+	}
+
 	if err := bus.Dispatch(&query); err != nil {
 		return nil, ApiError(404, "Dashboard not found", err)
 	}
 	return query.Result, nil
 }
 
 func DeleteDashboard(c *middleware.Context) Response {
-	dash, rsp := getDashboardHelper(c.OrgId, c.Params(":slug"), 0)
+	dash, rsp := getDashboardHelper(c.OrgId, c.Params(":slug"), 0, "")
```

Comment:
This wont work when we remove slug uniqueness.

Reply:
Fixed in frontend_step2 branch so will be fixed when merging in that

Reply:
Just a comment, having multiple branches going at the same time seems to be causing a lot of confusion. Is there a reason we're not just doing the dev in a single branch?

Reply:
At first we were unsure if we was going to be able to include all the needed changes in beta 1. So therefore we decided to split up the initial work to have the possibility to merge the initial work to beta 1. 

Second, after the initial work we had different tasks that could be divided between several people with possibility to work in parallel on multiple branches. Having multiple branches we could review them separately and decide to merge some of the work and include it in beta 1. 

One alternative could have been to use one branch and cherry pick commits to beta 1 branch, but then there's always a risk that you cannot cherry pick one commit because it's dependent on another that we may not want to hit beta 1.

The idea is still to review the branches using separate PR's and then merge them to this branch one at a time. Hope this make things more clearer,

Right now it looks promising for getting almost everything included in the beta 1.